Persistence.MongoDB/Serializers/InstantSerializer.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using MongoDB.Bson;
+using MongoDB.Bson.Serialization;
+using MongoDB.Bson.Serialization.Serializers;
+using NodaTime;
+
+namespace Persistence.MongoDB.Serializers
+{
+    public class InstantSerializer : SerializerBase<Instant>
+    {
+        public static readonly InstantSerializer Instance = new InstantSerializer();
+
+        private InstantSerializer()
+        {
+        }
+
+        public override Instant Deserialize(BsonDeserializationContext context, BsonDeserializationArgs args)
+        {
+            BsonType type = context.Reader.GetCurrentBsonType();
+            return type switch
+            {
+                BsonType.Null => default,
+                BsonType.DateTime => Instant.FromUnixTimeMilliseconds(context.Reader.ReadDateTime()),
+                _ => throw new NotSupportedException($"Cannot convert type '{type}' to Instant.")
```

Comment:
InvalidCastException? I suppose NotSupportedException is fine though

Reply:
MongoDB's `SerializerBase` actually offers two methods: `CreateCannotBeSerializedException` and `CreateCannotBeDeserializedException`, which throw `NotSupportedException` themselves, but apparently hide that as a detail. I'll just use `CreateCannotBeDeserializedException` here.